dev_guide/builds/build_inputs.adoc

```diff
@@ -458,6 +458,52 @@ section in your *_.gitconfig_* file:
 ----
 ====
 
+[[source-secrets-gitconfig-file-secured]]
+==== .gitconfig File for Secured Git
+
+If your Git server is secured with 2-way SSL and user name with password
+you must add the certificate files to your source build and add references to
+them in the *_.gitconfig_* file:
+
+. Add the *_client.crt_*, *_cacert.crt_*, and *_client.key_* files to the 
+*_/var/run/secrets/openshift.io/source/_* folder in the
+xref:../../dev_guide/application_lifecycle/new_app.adoc#specifying-source-code[application
+source code].
+
+. In the *_.gitconfig_* file for the application, add the `[http]` section
+shown in the following example:
++
+----
+# cat .gitconfig
+[user]
+        name = <name>
+        email = <email>
+[http]
+        sslVerify = false
+        sslCert = /var/run/secrets/openshift.io/source/client.crt
+        sslKey = /var/run/secrets/openshift.io/source/client.key
+        sslCaInfo = /var/run/secrets/openshift.io/source/cacert.crt
+----
+
+. Create the secret        
+----
+$ oc create secret generic <secret_name> \
+--from-literal=username=<user_name> \ <1>
+--from-literal=password=<password> \ <2>
+--from-file=.gitconfig=.gitconfig \
+--from-file=client.crt=/var/run/secrets/openshift.io/source/client.crt \
+--from-file=cacert.crt=/var/run/secrets/openshift.io/source/cacert.crt \
+--from-file=client.key=/var/run/secrets/openshift.io/source/client.key
+----
+<1> The user's Git user name.
+<2> The password for this user.
+
+[IMPORTANT]
+====
+To avoid having to enter your password, be sure to specify the S2I image in 
+your builds.
+====
```

Comment:
Suggest to make it more clear about this part: If your local git configuration is NOT able to git clone the repository, specifying the S2I image will avoid to promote asking you to provide username and password.